content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,236 @@
+
+The first time you clone an HTTPS URL that requires authentication, Git will prompt you to log in using a browser window.
+If you enabled two-factor authentication (like you should have), you'll need to complete the 2FA challenge.
+
+Once you've authenticated successfully, your credentials are stored in the Git Credential Manager and will be used every time you clone an HTTPS URL.
+Git will not ask you for your credentials again unless you change your credentials.
+
+In the event that you are not asked for an authentication via the browser, but instead are asked for your username and password, you will need to fill in your GitHub username and a PAT (not your GitHub password, which is indeed confusing).
+The next section explains how you do this (and how you avoid having to copy-paste a PAT manually).
+
+## Creating Personal Access Tokens (PAT)
+
+Personal access tokens can only be used for HTTPS Git operations.
+When you create a PAT, you will be asked which scopes should be allowed (i.e. what rights do you give).
+We suggest that you stick with the scopes recommended by the `usethis` package.
+Executing the following command takes your browser to a pre filled form with the recommended scopes (repo, user, workflow) where you can create your PAT.
+
+```{r eval=FALSE}
+usethis::create_github_token()
+```
+
+After you have done this, you can store this PAT in the Git Credential Manager using this code:
+
+```{r eval=FALSE}
+gitcreds::gitcreds_set()
+```
+
+which will open a prompt where you can paste your PAT (or replace an old one).
+
+If you want to know more about PATs read [this section from happy git with R](https://happygitwithr.com/https-pat.html#get-a-pat) and [this section from the usethis package](https://usethis.r-lib.org/articles/git-credentials.html#get-a-personal-access-token-pat).
+
+## Checking your authentication settings
+
+The `usethis` package has a function to get a situation report on your current Git/GitHub status, including information about authentication:
+
+```{r eval=FALSE}
+usethis::git_sitrep()
+```
+
+If you see warnings about git_vaccinated, it is recommended to execute
+
+```{r}
+usethis::git_vaccinate()
+```
+
+which adds a global `.gitignore` file that decreases the chance that you accidentally leak credentials.
+
+
+## <a name="ssh"></a>Creating public-private SSH-key pairs
+
+Below we describe the steps you need to run in order to make SSH-keys.
+The SSH protocol is, next to the HTTPS-protocol, a way to safely interact with 
+GitHub.
+A private key, that is specific to your computer, and a matching public key that
+is stored on your GitHub account are needed.
+The provided instructions are for Windows and are taken from
+https://happygitwithr.com/ssh-keys.html.
+The same source may be consulted for other operating systems.
+
+1. Open RStudio: _Tools > Global Options…> Git/SVN > Create RSA Key..._ You can
+optionally use a passphrase for extra protection of the key.
```

Comment:
Without password, anyone who has a copy of your private key, can impersonate you when authenticating on GitHub. When the private key is password protected they need your password too.